packages/teraslice/src/lib/cluster/services/cluster/backends/kubernetesV2/index.ts

```
@@ -42,8 +42,7 @@ export class KubernetesClusterBackendV2 {
             this.logger,
             null,
             kubernetesNamespace,
-            // @ts-expect-error
-            context.sysconfig.teraslice.kubernetes_api_poll_delay,
+            context.sysconfig.teraslice.kubernetes_api_poll_delay || 1000,
```

Is this needed? Usually default values and settings belong in the schema, it worries me a little bit that we are not relying on it. If its conditional on other settings we could rely on post validation to set it or something

---

This is one of those cases where the type can be undefined, but terafoundation validation will fill in a default. Maybe it is better to throw an error if this value is undefined? Or is there some better way to handle this that I'm missing?

---

I made this property non-optional on the teraslice Config type since it has a default in the schema.